Source/DotNET/Applications/Identity/IdentityProviderEndpoint.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Aksio Insurtech. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.AspNetCore.Http;
+
+namespace Aksio.Applications.Identity;
+
+/// <summary>
+/// Represents the actual endpoint called for identity details (/.aksio/me).
+/// </summary>
+public class IdentityProviderEndpoint
+{
+    readonly JsonSerializerOptions _serializerOptions;
+    readonly IProvideIdentityDetails _identityProvider;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="IdentityProviderEndpoint"/> class.
+    /// </summary>
+    /// <param name="serializerOptions"><see cref="JsonSerializerOptions"/> to use for serialization.</param>
+    /// <param name="identityProvider"><see cref="IProvideIdentityDetails"/> for providing the identity.</param>
+    public IdentityProviderEndpoint(JsonSerializerOptions serializerOptions, IProvideIdentityDetails identityProvider)
+    {
+        _serializerOptions = serializerOptions;
+        _identityProvider = identityProvider;
+    }
+
+    /// <summary>
+    /// Handle the identity request.
+    /// </summary>
+    /// <param name="request"><see cref="HttpRequest"/> that holds all the request information.</param>
+    /// <param name="response"><see cref="HttpResponse"/> that will be written to.</param>
+    /// <returns>Awaitable task.</returns>
+    public async Task Handler(HttpRequest request, HttpResponse response)
```

Comment:
Just a style preference but I like to turn more complex conditionals into predicate functions.

So instead of checking all the request headers individually, a function called something like HasValidIdentityHeaders().  Also something further down like "IsValidJsonToken".

For a more simple method with a more technical role, it's perfectly ok to keep everything inline.

Reply:
Good point!

Reply:
I agree to this, and also revert the test to reduce nesting